lib/ckb/wallet.rb

```diff
@@ -137,20 +151,28 @@ def generate_tx(target_address, capacity, data = "0x", key: nil, fee_rate: 0, us
     # @param capacity [Integer]
     # @param data [String] "0x..."
     # @param key [CKB::Key | String] Key or private key hex string
-    # @param fee_rate [Integer] shannons per KB
-    def send_capacity(target_address, capacity, data = "0x", key: nil, fee_rate: 0)
-      tx = generate_tx(target_address, capacity, data, key: key, fee_rate: fee_rate)
+    # @param fee [Integer] shannons
+    # @param fee_rate [Integer] shannons per KB, using fee first
+    def send_capacity(target_address, capacity, data = "0x", key: nil, fee: 0, fee_rate: 0)
+      tx = generate_tx(target_address, capacity, data, key: key, fee: fee, fee_rate: fee_rate)
       send_transaction(tx)
     end
 
     # @param capacity [Integer]
     # @param key [CKB::Key | String] Key or private key hex string
+    # @param fee [Integer] shannons
     # @param fee_rate [Integer] shannons per KB
     #
     # @return [CKB::Type::OutPoint]
-    def deposit_to_dao(capacity, key: nil, fee_rate: 0)
+    def deposit_to_dao(capacity, key: nil, fee: 0, fee_rate: 0)
       key = get_key(key)
 
+      mode = if fee == 0 && fee_rate > 0
+               :fee_rate
+             else
```

Comment:
may be you can extract a separate method

Reply:
updated